### PR TITLE
HHH-13061 Introduce a 'none' option for BytecodeProvider implementati…

### PR DESCRIPTION
**Pull request by Sanne**
…on choice

As proposed on the mailing list.

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/bytecode/internal/none/BytecodeProviderImpl.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			Class clazz,
+			String[] getterNames,
+			String[] setterNames,
+			Class[] types) {
+		throw new HibernateException( "Using the ReflectionOptimizer is not possible when the configured BytecodeProvider is 'none'. Disable " + AvailableSettings.USE_REFLECTION_OPTIMIZER + " or use a different BytecodeProvider");
+	}
+
+	@Override
+	public Enhancer getEnhancer(EnhancementContext enhancementContext) {
+		return null;
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/bytecode/internal/none/BytecodeProviderImpl.java]: Just a minor nitpick but shouldn't we throw an exception here too?

**Comment by gsmet** [comment]
So, from what I understand, if you want to use it in a normal setup, you would have to use a real bytecode provider at compile time and this one at runtime?

**Comment by Sanne** [comment]
> So, from what I understand, if you want to use it in a normal setup, you would have to use a real bytecode provider at compile time and this one at runtime?

Exactly.

File: hibernate-core/src/main/java/org/hibernate/bytecode/internal/none/BytecodeProviderImpl.java (hunk 1 above)
Comment (Sanne) [reply hibernate-core/src/main/java/org/hibernate/bytecode/internal/none/BytecodeProviderImpl.java]: No the null value is actually what the API clients are expecting.

**Comment by gsmet** [comment]
Merged. I don't think we should advertise it for now so let's not document it.

**Pull request closed without merging**